source/javascripts/hooks/useEnvVars.ts

```diff
@@ -0,0 +1,141 @@
+import { useMemo } from 'react';
+import { useShallow } from 'zustand/react/shallow';
+import { useQueries, useQuery } from '@tanstack/react-query';
+import useBitriseYmlStore from '@/hooks/useBitriseYmlStore';
+import StepService from '@/core/models/StepService';
+import { EnvVar } from '@/core/models/EnvVar';
+import EnvVarService from '@/core/models/EnvVarService';
+import EnvVarsApi from '@/core/api/EnvVarsApi';
+import WindowUtils from '@/core/utils/WindowUtils';
+import StepApi from '@/core/api/StepApi';
+import WorkflowService from '@/core/models/WorkflowService';
+
+const useDefaultEnvVars = () => {
+  const appSlug = WindowUtils.appSlug() ?? '';
+  const projectType = useBitriseYmlStore(useShallow((s) => s.yml.project_type)) as never;
+
+  return useQuery({
+    enabled: Boolean(appSlug),
+    queryKey: ['default-env-vars', appSlug, projectType],
+    queryFn: ({ signal }) => EnvVarsApi.getEnvVars({ appSlug, projectType, signal }),
+  });
+};
+
+const useAppLevelEnvVars = () => {
+  return useBitriseYmlStore(
+    useShallow((s) => {
+      const envVarMap = new Map<string, EnvVar>();
+
+      s.yml.app?.envs?.forEach((envVarYml) => {
+        const env = EnvVarService.parseYmlEnvVar(envVarYml, 'app');
+        envVarMap.set(env.key, env);
+      });
+
+      return Array.from(envVarMap.values());
+    }),
+  );
+};
+
+const useWorkflowLevelEnvVars = (workflowId: string) => {
+  return useBitriseYmlStore(
+    useShallow((s) => {
+      const envVarMap = new Map<string, EnvVar>();
+
+      WorkflowService.getBeforeRunChain(s.yml.workflows ?? {}, workflowId).forEach((id) => {
+        s.yml.workflows?.[id]?.envs?.forEach((envVarYml) => {
+          const env = EnvVarService.parseYmlEnvVar(envVarYml, id);
+          envVarMap.set(env.key, env);
```

Comment:
Can we add a comment for a later follow-up?

For the correct availability order, should it be the following later on:
- WF envs from the first before_run
- Step envs from the steps of the first before run
...
- WF envs from the Nth before_run
- Step envs from the steps of the Nth before run
- WF envs from the Workflow with workflowId
- Step envs from the steps of the Workflow with workflowId